paymentapi/src/main/java/fi/hel/verkkokauppa/payment/api/PaytrailPaymentController.java

```diff
@@ -181,4 +181,35 @@ public ResponseEntity<TreeMap<String, String>> getUpdateCardFormParameters(
             throw new CommonApiException(HttpStatus.INTERNAL_SERVER_ERROR, error);
         }
     }
+
+    @PostMapping("/payment/paytrail/check-card-update-return-url")
+    public ResponseEntity<Void> checkCardUpdateReturnUrl(
+            @RequestBody GetPaymentRequestDataDto dto,
+            @RequestParam Map<String, String> params,
+            @RequestParam(value = "signature") String signature,
+            @RequestParam(value = "checkout-tokenization-id") String tokenizationId
+    ) {
+        try {
+            String namespace = dto.getOrder().getOrder().getNamespace();
+            String merchantId = "";
+            if (dto.getOrder().getItems() != null && dto.getOrder().getItems().size() > 0 && dto.getOrder().getItems().get(0).getMerchantId() != null) {
+                merchantId = dto.getOrder().getItems().get(0).getMerchantId();
```

Comment:
You should create helper function to get merchant is from order, we have service/util class functionality in controller now.